core/field.js

```diff
@@ -852,7 +852,10 @@ Blockly.Field.prototype.setValue = function(newValue) {
     return;
   }
 
-  if (this.sourceBlock_ && Blockly.Events.isEnabled()) {
+  if (this.sourceBlock_ &&
+      !this.sourceBlock_.disposed &&
```

Comment:
Is this a change we need no matter what?

Reply:
Ah, yep I forgot I wanted to do that. Pull request at #3766